Makefile

```diff
@@ -121,6 +123,28 @@ endif
 .PHONY: update-embedded-full-documentation
 update-embedded-full-documentation: doc
 	rsync -a --delete doc/html/ lxd/gendocs/html/
+	# Remove the .doctrees directory, as it is useless for us and will reduce the size.
+	rm -Rf lxd/gendocs/html/.doctrees
+
+	# Minify content
+	@echo "Minifying HTML files..."
+	@find lxd/gendocs/html/ -name '*.html' -type f -print0 | xargs -0 -n 1 -P `nproc` -I {} $(HTMLMINIFY) -o {} {}
+
+	@echo "Minifying CSS files..."
+	@find lxd/gendocs/html/ -name '*.css' -type f -print0 | xargs -0 -n 1 -P `nproc` -I {} postcss --use cssnano -o {} {}
+
+	@echo "Minifying JS files..."
+	@find lxd/gendocs/html/ -name '*.js' -type f -print0 | xargs -0 -n 1 -P `nproc` -I {} $(JSMINIFY) -o {} {}
+
+	@echo "Minifying SVG files..."
+	@find lxd/gendocs/html/ -name '*.svg' -type f -print0 | xargs -0 -n 1 -P `nproc` -I {} svgo {} --output={}
+
+	@echo "Optimizing PNG files..."
+	@find lxd/gendocs/html/ -name '*.png' -type f -print0 | xargs -0 -n 1 -P `nproc` -I {} optipng -o7 {}
+
+	# Compressing content
+	@echo "Compressing files..."
+	@find lxd/gendocs/html/ -type f \( -not -name "*.mp3" -and -not -name "*.wav" -and -not -name "*.ogg" \) -print0 | xargs -0 -n 1 -P `nproc` -I {} gzip {} -9
```

Comment:
If the HTML, CSS, JS and SVG files are compressed inline (see suggestion above), what else would that compress? If that final compression pass stays, I think it'd be safe to exclude `*.gz` filenames to avoid double compression.